src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +36,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+            func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7AEC.s")
+    if (this->unk_216 < 30) {
+        func_808B7770(this, globalCtx, 0.5f);
+    } else {
+        func_808B7770(this, globalCtx, 0.3f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7AFC.s")
+    tempUnk210 = (this->unk_210 - 500) * 0.0006f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7B58.s")
+    clampedTempUnk210 = CLAMP(tempUnk210, 0.0f, 1.5f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7B6C.s")
+    func_800F436C(&this->dyna.actor.projectedPos, 203, clampedTempUnk210);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7BB0.s")
+void func_808B818C(BgSpot18Basket* this) {
+    this->actionFunc = func_808B81A0;
+    this->unk_216 = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7BCC.s")
+void func_808B81A0(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    s32 i;
+    Actor* actor = &this->dyna.actor;
+    Vec3f tempVector;
+    EnItem00* collectible;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7D38.s")
+    if (this->unk_216 == 1) {
+        tempVector.x = actor->posRot.pos.x;
+        tempVector.y = actor->posRot.pos.y + 170.0f;
+        tempVector.z = actor->posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7D50.s")
+        if (this->unk_218 == 0) {
+            for (i = 0; i < ARRAY_COUNT(D_808B85E4); i++) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 4);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[i];
+                }
+            }
+        } else if (this->unk_218 == 1) {
+            for (i = 0; i < ARRAY_COUNT(D_808B85E4); i++) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 0);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[i];
+                }
+            }
+        } else if (this->unk_218 == 2) {
+            if ((this->unk_21A != 0) || (Flags_GetCollectible(globalCtx, (actor->params & 0x3F)) != 0)) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 20);
+                if (collectible != NULL) {
+                    collectible->actor.velocity.y = 11.0f;
+                    collectible->actor.posRot.rot.y = D_808B85E4[1];
+                }
+            } else {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, ((actor->params & 0x3F) << 8) | 6);
```

Comment:
I'll show this one since it's a bit more complex than the others but it also applies to part of the argument here (the other part seems to be the collectible flag to toggle when picking up the heart piece).
```suggestion
                collectible = Item_DropCollectible(globalCtx, &tempVector, ((actor->params & 0x3F) << 8) | ITEM00_HEART_PIECE);
```